Enable strict mypy checks

ludeeus opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Proposed change

<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->

Enable strict mypy checks
replaces #53

## Type of change

<!--
  What type of change does your PR introduce?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionalit)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Additional information

<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass.
- [x] There is no commented out code in this PR.
- [x] The code has been formatted using Black (`make black`)
- [x] Tests have been added to verify that the new code works.

ludeeus commented:
@KapJI can you take a look at this?

KapJI commented on awesomeversion/awesomeversion.py:
```diff
@@ -14,17 +15,18 @@
         str.__init__(string)
 
-    def __new__(cls, version, _=None):
+    def __new__(cls, version: str, _: Optional[Any] = None) -> "_AwesomeVersionBase":
```
You can add `from __future__ import annotation` to use unquoted type.
```suggestion
    def __new__(cls, version: str, _: Optional[Any] = None) -> _AwesomeVersionBase:
```

ludeeus replied:
`from __future__ import annotation` can only be used from python 3.8, this library has support for 3.6 and 3.7 as well.

KapJI reviewed (approved):
LGTM, thanks!

In #53 I also added basic isort config, it might be useful to have consistent output for isort and black.

ludeeus commented:
> In #53 I also added basic isort config, it might be useful to have consistent output for isort and black.

 I did not see that, added that as well now :+1:

ludeeus commented:
Thanks for looking it over @KapJI :+1:
